### PR TITLE
Add full scoring support

### PR DESCRIPTION
**Pull request by chrisroberts**
Builds out full version scoring support for ideal "least impact" updates (as described in #11)

### CODE REVIEW

File: lib/batali/score_keeper.rb [hunk 1]
```diff
@@ -6,12 +6,53 @@ class ScoreKeeper < Grimoire::UnitScoreKeeper
...
+            multiplier = multi_val + (multi_val * distance)
+          end
+        end
+      end
+      score = []
+      unit.version.segments.reverse.each_with_index.map do |val, pos|
+        if(val == 0)
+          score.push 0
+        else
+          score.push (2 - (1.0 / val)) * ((pos + 1)**10)
```
Comment (ckuttruff) [inline lib/batali/score_keeper.rb]: Can we add a quick comment about what the math is doing here?

Comment (chrisroberts) [inline lib/batali/score_keeper.rb]: "magic" :sparkles:

Comment (ckuttruff) [inline lib/batali/score_keeper.rb]: lol, indeed

**Comment by chrisroberts** [comment]
@ckuttruff added notes!

File: lib/batali/score_keeper.rb (hunk 1 above)
Comment (ckuttruff) [inline lib/batali/score_keeper.rb]: sweet, thanks for the additional comments

**Pull request merged**